tests/django_restframework_gis_tests/tests.py

```diff
@@ -584,3 +584,58 @@ def test_pickle(self):
         pickled = pickle.dumps(geojsondict)
         restored = pickle.loads(pickled)
         self.assertEqual(restored, geojsondict)
+
+    def test_geometrycollection_geojson(self):
+        """ test geometry collection geojson behaviour """
+        location = Location.objects.create(name='geometry collection geojson test',
+                                           geometry='GEOMETRYCOLLECTION ('
+                                                    'POINT (135.0 45.0),'
+                                                    'LINESTRING (135.0 45.0,140.0 50.0,145.0 55.0),'
+                                                    'POLYGON ((135.0 45.0,140.0 50.0,145.0 55.0,135.0 45.0)))')
+        url = reverse('api_geojson_location_details', args=[location.id])
+        expected = {
+            'id': location.id,
+            'type': 'Feature',
+            'properties': {
+                'details': "http://testserver/geojson/%s/" % location.id,
+                'name': 'geometry collection geojson test',
+                'fancy_name': 'Kool geometry collection geojson test',
+                'timestamp': None,
+                'slug': 'geometry-collection-geojson-test',
+            },
+            'geometry': {
+                'type': 'GeometryCollection',
+                'geometries': [
+                    {
+                        'type': 'Point',
+                        'coordinates': [135.0, 45.0]
+                    },
+                    {
+                        'type': 'LineString',
+                        'coordinates': [
+                            [135.0, 45.0],
+                            [140.0, 50.0],
+                            [145.0, 55.0]
+                        ]
+                    },
+                    {
+                        'type': 'Polygon',
+                        'coordinates': [
+                            [
+                                [135.0, 45.0],
+                                [140.0, 50.0],
+                                [145.0, 55.0],
+                                [135.0, 45.0],
+                            ]
+                        ]
+                    },
+                ],
+            }
+        }
+        response = self.client.get(url)
+        if sys.version_info > (3, 0, 0):
+            self.assertCountEqual(json.dumps(response.data), json.dumps(expected))
+        else:
+            self.assertItemsEqual(json.dumps(response.data), json.dumps(expected))
+        self.assertContains(response, "Kool geometry collection geojson test")
+        self.assertJSONEqual(json.dumps(response.data), expected)
```

Comment:
Sorry, this was a misunderstanding on my part thinking that assertCountEqual and assertItemsEqual wouldn't do a thorough job because of their names, but on more research I see that they work very well.  Line 640 is there because it came with the tests I copied and pasted from test_geojson_format.  I have no problem removing it as well.

Removed line 641.